build/configs/artik053/Make.defs

```diff
@@ -111,12 +117,31 @@ endif
 
 ARCHCFLAGS = -fno-builtin -mcpu=cortex-r4 -mfpu=vfpv3
 ARCHCXXFLAGS = -fno-builtin -mcpu=cortex-r4 -mfpu=vfpv3
+ifneq ($(CONFIG_HAVE_CXX),y)
+ARCHCXXFLAGS += -fexceptions
+endif
 ifeq ($(QUICKBUILD),y)
 ARCHWARNINGS = -Wall -Werror -Wstrict-prototypes -Wshadow -Wundef -Wno-implicit-function-declaration -Wno-unused-function -Wno-unused-but-set-variable
-ARCHWARNINGSXX = -Wall -Wshadow -Wundef
+ARCHWARNINGSXX = -Wall 
+ifeq ($(CONFIG_HAVE_CXX),y)
+ARCHWARNINGSXX += -Wshadow -Wundef
+
+else ifeq ($CONFIG_GRPC),y)
+ARCHWARNINGSXX += -Wno-sign-compare
+else
+ARCHWARNINGSXX += -Werror
+endif
```

Comment:
This looks strange.
When CONFIG_HAVE_CXX is enabled, will not CONFIG_GRPC be enabled? I think they are independent.